client/components/Signup.tsx

```diff
@@ -28,19 +40,21 @@ function Signup() {
 
     const navigate = useNavigate();
 
-    const switchForm = () => {
-        navigate('/login');
-    };
+    const [signupMutation] = useMutation(SIGNUP_MUTATION, {
+        onCompleted: (data) => {
+            setAuthContext(data.signup, data.signup.token);
+            navigate('/dashboard');
+        },
+        onError: (error) => console.log(error),
+    });
 
-    const handleClick = (e: any) => {
+    const handleClick = async (e: any, userData: ISState['user']) => {
         e.preventDefault();
-        navigate('/dashboard');
-        // console.log(user);
+        signupMutation({ variables: { user: userData } });
```

Comment:
Where is signupMutation coming from? 
Is it from line 43? Why the " [ ] " around it?

Reply:
It was original Apollo syntax. I just removed it and its ok with out the `[  ]`

Reply:
Update. The `[ ]` are required. It's just apollo syntax.